addons/website_sale_stock/models/product_template.py

```diff
@@ -78,3 +78,12 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
             })
 
         return combination_info
+
+    def _is_sold_out(self):
+        combination_info = self.with_context(website_sale_stock_get_quantity=True)._get_combination_info()
+        return combination_info['product_type'] == 'product' and combination_info['free_qty'] <= 0
+
+    def _get_website_ribbon(self):
+        if self._is_sold_out():
+            return self.env['product.ribbon'].search([('id', '=', '2')])
```

Comment:
I doubt `('id', '=', '2')` is a stable reference to the ribbon you want.
You can access pre-defined records based on their xmlid:
```py
self.env.ref('website_sale.sold_out_ribbon');
```